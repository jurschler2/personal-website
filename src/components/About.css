/** 
 * Styling for the About component:
 */

.about-contents {
  display: flex;
  margin-left: 15%;
  width: 100%;
  height: 25%;
}
.bullet-points-container {
  flex: 1;
  margin-left: 200%;
}
.img-headshot {
  width: 250px;
  border-radius: 2px;
  flex: 2;
  margin-left: 400%;
}
.about-header {
  font-family: Kanit;
  font-weight: 300;
  font-size: 150%;
  color: #7b7b87;
  display:  block;
  margin-left: 30%;
  text-align: left;
  padding: 2%;
}