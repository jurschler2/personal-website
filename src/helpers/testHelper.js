import React from "react";
import { render } from "@testing-library/react";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

export function renderWithRouter(children, initialPath = "") {
  const history = createMemoryHistory();
  if (initialPath) history.push(initialPath);
  return render(<Router history={history}>{children}</Router>);
}
