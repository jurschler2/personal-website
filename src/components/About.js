import React from "react";
import BulletPoint from "./BulletPoint";
import { points, images } from "../content/data";
import "./About.css";

/**
 *  DESCRIPTION:
 *  PROPS: None
 *  FLOW: App => Routes => About
 *  PARENT: Routes
 *  CHILDREN: BulletPoint
 */

function About() {


  const renderBulletPoints = () => {

    return points.map(p => <BulletPoint point={p} />)
  }

  return (
  <>  
    <div className="about-header">
      Here is a little about me:
    </div>
    <div className="about-contents flex-container">
      <div className="bullet-point-container">
        {renderBulletPoints()}
      </div>
      <div className="img-container">
        <img className="img-headshot" src={images.headshot.source} alt=""/>
      </div>
    </div>
  </>
  )
}

export default About;