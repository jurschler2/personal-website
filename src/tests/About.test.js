import React from "react";
import About from "../components/About";
import {renderWithRouter} from "../helpers/testHelper"

it("renders without crashing", () => {
  renderWithRouter(<About />);
});

it("matches snapshot", () => {
  const { asFragment } = renderWithRouter(<About />);
  expect(asFragment()).toMatchSnapshot();
});