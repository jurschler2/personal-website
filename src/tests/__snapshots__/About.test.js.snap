// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="about-header"
  >
    Here is a little about me:
  </div>
  <div
    class="about-contents flex-container"
  >
    <div
      class="bullet-point-container"
    >
      <div
        class="point 1 "
      >
        This is point 1.
      </div>
      <div
        class="point 2 "
      >
        This is point 2.
      </div>
      <div
        class="point 3 "
      >
        This is point 3.
      </div>
    </div>
    <div
      class="img-container"
    >
      <img
        alt=""
        class="img-headshot"
        src="headshot.png"
      />
    </div>
  </div>
</DocumentFragment>
`;
